client/lib/reader-post-flux-adapter/index.js

```diff
@@ -42,6 +42,16 @@ const fluxPostAdapter = Component => {
 		state = this.getStateFromStores( this.props );
 
 		updateState = ( newState = this.getStateFromStores() ) => {
+			// check to see if this new state is the same as the old state
+			if ( newState.posts.length === this.state.posts.length ) {
```

Comment:
> this avoids needing shouldComponentUpdate entirely. By not calling setState, we never have to deal with reconciliation.
  
shouldComponentUpdate is _specifically_ for avoiding reconciliation: https://facebook.github.io/react/docs/optimizing-performance.html#avoid-reconciliation

> shouldComponentUpdate is for dealing with props that someone else is passing you. this is for dealing with internal state you're setting on yourself.

the reason shouldComponentUpdate takes in _both_ props and state is so that it can be used for both

> and yes, it would still avoid the render if we had it in shouldComponentUpdate. But this avoids needing that at all.

Sure, in the end it makes no difference.  The reason I like putting it in shouldComponentUpdate is because its so much more explicit.

- shouldComponentUpdate answers the question of whether or not a component should update
- updateState updates the state

when you put both in updateState it loses that separation of concerns.
It won't block me giving a 🚢  at all though :)